### PR TITLE
Remove RemixerInstance.

### LINKED ISSUE #54: Remove instances of RemixerInstance
Do a project-wide search for "RemixerInstance". It still appears in import statements and in RemixerBindingException error messages.

### LINKED ISSUE #55: RemixerBinder.bind(this) fails if Activity does not have any @VariableMethod annotations
We have two choices:

1. Throw a better exception with a good error message.
1. Soft-fail instead.

### PR DESCRIPTION
**Pull request by miguelandres**
It is already fully deprecated and adding confusion.

Fixes #54 

Fixes #55

### CODE REVIEW

File: remixer_core/src/main/java/com/google/android/libraries/remixer/annotation/RemixerBindingException.java [hunk 1]
```diff
@@ -26,12 +26,11 @@
    * Advice to append at the end of messages.
    */
   private static final String ADVICE =
-      ". Make sure that you're using @RemixerInstance on an instance of this class and that you"
-          + " are not proguarding out this class' binder. Normally adding this to your proguard"
-          + " configuration should help: "
+      ". Make sure that you're using com.google.android.libraries.remixer.annotation. *Method"
```
Comment (pingpongboss) [inline remixer_core/src/main/java/com/google/android/libraries/remixer/annotation/RemixerBindingException.java]: do you mean to have a space in front of `*Method`?

File: remixer_core/src/main/java/com/google/android/libraries/remixer/annotation/RemixerBindingException.java [hunk 2]
```diff
@@ -26,12 +26,11 @@
...
+          + " annotations in this class, are not proguarding out this class' binder. Normally"
+          + " adding this to your proguard configuration should help: "
```
Comment (pingpongboss) [inline remixer_core/src/main/java/com/google/android/libraries/remixer/annotation/RemixerBindingException.java:31]: Also just FYI you should be able to define this for clients via `consumerProguardFiles` in your library's `build.gradle` file.

**Review by pingpongboss — APPROVED** [review]

File: remixer_core/src/main/java/com/google/android/libraries/remixer/annotation/RemixerBindingException.java (hunk 1 above)
Comment (miguelandres) [reply remixer_core/src/main/java/com/google/android/libraries/remixer/annotation/RemixerBindingException.java]: Nope, good catch

File: remixer_core/src/main/java/com/google/android/libraries/remixer/annotation/RemixerBindingException.java (hunk 2 above)
Comment (miguelandres) [reply remixer_core/src/main/java/com/google/android/libraries/remixer/annotation/RemixerBindingException.java:31]: It should be done already. just paranoia.

**Pull request merged**